crates/conjure_core/src/rules/constant.rs

```diff
@@ -81,15 +81,50 @@ pub fn eval_constant(expr: &Expr) -> Option<Lit> {
                 return None;
             }
 
-            Some(Lit::Bool(a / b == c))
+            let a = a as f32;
+            let b = b as f32;
+            let div: i32 = (a / b).floor() as i32;
+            Some(Lit::Bool(div == c))
         }
         Expr::Bubble(_, a, b) => bin_op::<bool, bool>(|a, b| a && b, a, b).map(Lit::Bool),
 
         Expr::Reify(_, a, b) => bin_op::<bool, bool>(|a, b| a == b, a, b).map(Lit::Bool),
-        _ => {
-            warn!(%expr,"Unimplemented constant eval");
-            None
+        Expr::SumEq(_, exprs, a) => {
+            flat_op::<i32, bool>(|e, a| e.iter().sum::<i32>() == a, exprs, a).map(Lit::Bool)
+        }
+        Expr::ModuloEqUndefZero(_, a, b, c) => {
+            // modulo always rounds down
+            let a = unwrap_atom::<i32>(a)?;
+            let b = unwrap_atom::<i32>(b)?;
+            let c = unwrap_atom::<i32>(c)?;
+
+            if b == 0 {
+                return None;
+            }
+
+            let modulo = (a as f32 % b as f32).floor() as i32;
```

Comment:
this is unlikely to be correct, say for negative numbers? SR docs suggest `a-b*floor(a/b)`

Reply:
I added this, but it breaks the mod_02 test:

```
find a : int(0..3)
such that a >= 4 % 2 $ should be simplified to constant
```

```
Diff < left / right > :
<Array []
>Array [
>    Object {
>        "a": Number(0),
>    },
>    Object {
>        "a": Number(1),
>    },
>    Object {
>        "a": Number(2),
>    },
>    Object {
>        "a": Number(3),
>    },
>
```

Reply:
I see my mistake !

Reply:
hmmmm, I'll look into it.

Reply:
I fixed it: I did (a-b) * floor(a/b) but it should be a - b * floor(a/b)

Reply:
I was getting worried :) Good to see the testing infrastructure is doing its job though!